jenkins_pipelines/environments/common/pipeline-build-validation.groovy

```diff
@@ -74,181 +70,184 @@ def run(params) {
                     deployed = true
                 }
             }
+            /** Deploy stage end **/
 
+            /** Sanity check stage - always run **/
             stage('Sanity check') {
                 sh "./terracumber-cli ${common_params} --logfile ${resultdirbuild}/testsuite.log --runstep cucumber --cucumber-cmd 'cd /root/spacewalk/testsuite; ${env.exports} rake cucumber:build_validation_sanity_check'"
             }
 
-            stage('Run core features') {
-                if (params.must_run_core && (deployed || !params.must_deploy)) {
+            /** Core features stage **/
+            if (params.must_run_core && (deployed || !params.must_deploy)) {
+                stage('Run core features') {
                     sh "./terracumber-cli ${common_params} --logfile ${resultdirbuild}/testsuite.log --runstep cucumber --cucumber-cmd '${env.exports} cd /root/spacewalk/testsuite; rake cucumber:build_validation_core'"
                 }
             }
-
-            stage('Sync. products and channels') {
-                if (params.must_sync && (deployed || !params.must_deploy)) {
+            
+            /** Reposync stage **/
+            if (params.must_sync && (deployed || !params.must_deploy)){
+                stage('Sync. products and channels') {
                     // Get minion list from terraform state list command
                     def nodesHandler = getNodesHandler()
                     res_products = sh(script: "./terracumber-cli ${common_params} --logfile ${resultdirbuild}/testsuite.log --runstep cucumber --cucumber-cmd 'unset ${nodesHandler.envVariableListToDisable.join(' ')}; ${env.exports} cd /root/spacewalk/testsuite; rake cucumber:build_validation_reposync'", returnStatus: true)
```

Comment:
If the issue is related to reposync failing, we probably only need to change this for : 
sh "./terracumber-cli ${common_params} --logfile ${resultdirbuild}/testsuite.log --runstep cucumber --cucumber-cmd 'unset ${nodesHandler.envVariableListToDisable.join(' ')}; ${env.exports} cd /root/spacewalk/testsuite; rake cucumber:build_validation_reposync'"